src/pages/ExamsPage/components/ResetExamAttemptModal.jsx

```diff
@@ -45,27 +57,30 @@ const ResetExamAttemptButton = ({ username, examName, attemptId }) => {
         <ModalDialog.Footer>
           <ActionRow>
             <ModalDialog.CloseButton variant="tertiary">
-              {formatMessage(messages.ResetExamAttemptButtonCancel)}
+              {formatMessage(messages.ResetExamAttemptModalCancel)}
             </ModalDialog.CloseButton>
-            <Button
+            <StatefulButton
+              state={resetButtonStatus}
+              {...ResetButtonProps}
               variant="primary"
               onClick={e => { // eslint-disable-line no-unused-vars
+                setResetButtonStatus('pending');
```

Comment:
Could you post a demo of this, please? Just reading the code, it's not clear to me when the status is set to `complete` (or any other state). Since the component is controlled, and we pass the state that the button is in via a prop, don't we need to set the status explicitly to `complete` (or any other state), when the reset call is complete? How does this work as-is?

Reply:
Maybe a brief comment on what's updating this? I know I'd be confused by where it goes from pending to complete just by looking at the code.

Reply:
I don't think it is actually updating the state to `completed`. It looks like because the list of attempts is refreshed after a reject/verify action is successful, the entire `AttemptList` component re-renders, and the modal closes for that reason, but I can't find where we're updating the button state otherwise.

Reply:
Added a quick comment on how this works in both modal files.